underfs/swift/src/main/java/alluxio/underfs/swift/SwiftUnderFileSystem.java

```diff
@@ -46,21 +46,26 @@
 import javax.annotation.concurrent.ThreadSafe;
 
 /**
- * OpenStack Swift {@link UnderFileSystem} implementation based on the JOSS library. This
- * implementation does not support the concept of directories due to Swift being an object store.
- * All mkdir operations will no-op and return true and empty directories will not exist.
- * Directories with objects inside will be inferred through the prefix.
+ * OpenStack Swift {@link UnderFileSystem} implementation based on the JOSS library.
+ * mkdir operations create zero-byte objects ending with the folder suffix.
  */
 //TODO(calvin): Reconsider the directory limitations of this class.
+//TODO(adit): Abstract out functionality common with other object under storage.
 @ThreadSafe
 public class SwiftUnderFileSystem extends UnderFileSystem {
   private static final Logger LOG = LoggerFactory.getLogger(Constants.LOGGER_TYPE);
 
-  /** Suffix for an empty file to flag it as a directory. */
-  private static final String FOLDER_SUFFIX = "_$folder$";
+  /** Value used to indicate nested structure in Swift. */
+  private static final char PATH_SEPARATOR_CHAR = '/';
```

Comment:
it will be backwards in-compatible. if we decide to change it in the future, directories (or files in directories) will be inaccessible across versions. it will require a script to rename objects (which is a copy operation for swift) in the store manually; potentially a very heavy operation.

is there a reason we might want to allow '/' in the object name?